### DISCUSSION TITLE
De-dups after `ekg-global-rename-tag'?

### DISCUSSION THREAD

**Discussion by qingshuizheng** [ahyatt/ekg#111] [General]
Hi @ahyatt ,

There are duplicate tags if the new tag name already exist in the note. Do we wanna do some cleanup? 

For instance, a note with 2 tags: 
```
"test 1"  ; with (:index 0) in the properties column,  
"test 2"  ; with (:index 1)
```

After renaming "test 2" to "test 1"
```
"test 1" ; (:index 0) 
"test 1" ; (:index 1)
```

Since the indexes are different, they will co-exist until the note is open and save again.

![image](https://github.com/ahyatt/ekg/assets/78839107/92c96e8c-8dd4-43cc-86e3-d010fa36a89b)
![image](https://github.com/ahyatt/ekg/assets/78839107/beca9eaf-927a-4413-8e26-9da80ce425f0)

Zheng

**Comment by ahyatt** [comment]
Thanks, great find!  I fixed this in the latest `develop` commit, please let me know if it works.  In particular, try `ekg-clean-db` and it should fix this (and not cause the problem again on next rename).

**Reply by qingshuizheng** [reply]
Thanks! I'll check and see.

**Reply by qingshuizheng** [reply]
Confirmed it works, thanks!

**Comment by qingshuizheng** [comment]
https://github.com/ahyatt/ekg/blob/316798112e7423527bbb4d9a36f659f1c6cbdf83/ekg.el#L1351-1372

BTW, there's one thing I forgot to mention, `ekg-global-rename-tag` currently does not standardize the `to-tag` value, e.g. downcase, string-trim, comma replacement, etc (which is what `ekg--normalize-note` does).

Maybe change the interactive form from

```elisp
(interactive (list (completing-read "From tag: " (ekg-tags))
                     (completing-read "To tag: " (ekg-tags))))
```

to 

```elisp
(interactive (list (completing-read "From tag: " (ekg-tags))
                     (string-trim (downcase (string-replace "," "" (completing-read "To tag: " (ekg-tags)))))))
```

Hope you don't mind I mention it here.

Zheng

**Comment by ahyatt** [comment]
Good idea, I've done it (a slightly different way)!

**Reply by qingshuizheng** [reply]
Works! Thanks!
